tools/opt/opt.cpp

```diff
@@ -463,22 +470,24 @@ int main(int argc, const char** argv) {
     return 1;
   }
 
-  // Let's do validation first.
-  spv_context context = spvContextCreate(target_env);
-  spv_diagnostic diagnostic = nullptr;
-  spv_const_binary_t binary_struct = {binary.data(), binary.size()};
-  spv_result_t error =
-      spvValidateWithOptions(context, options, &binary_struct, &diagnostic);
-  if (error) {
-    spvDiagnosticPrint(diagnostic);
+  if (!skip_validator) {
+    // Let's do validation first.
+    spv_context context = spvContextCreate(target_env);
+    spv_diagnostic diagnostic = nullptr;
+    spv_const_binary_t binary_struct = {binary.data(), binary.size()};
+    spv_result_t error =
+        spvValidateWithOptions(context, options, &binary_struct, &diagnostic);
+    if (error) {
```

Comment:
Consider
```
  if (error) {
      spvDiagnosticPrint(diagnostic);
  }
  spvDiagnosticDestroy(diagnostic);
  spvValidatorOptionsDestroy(options);
  spvContextDestroy(context) 
  return error;
```

Reply:
The way the diff looks is confusing, but we want to `return error` only if there was an error.  We would have to do

```
if (error) {
  spvDiagnosticPrint(diagnostic);
}
  spvDiagnosticDestroy(diagnostic);
  spvValidatorOptionsDestroy(options);
  spvContextDestroy(context) 
if (error) {
  return error;
}
```
I think the extra `if` is not very nice, and it only saves 1 line of code.

Reply:
Sorry, I thought it was the end of the function.